price-feeder/oracle/oracle.go

```diff
@@ -287,14 +290,21 @@ func (o *Oracle) getOrSetProvider(providerName string) provider.Provider {
 		case config.ProviderHuobi:
 			priceProvider = provider.NewHuobiProvider()
 
+		case config.ProviderOkx:
+			okxProvider, err := provider.NewOkxProvider(context.TODO(), o.providerPairs[config.ProviderOkx]...)
+			if err != nil {
+				return nil, err
+			}
+			priceProvider = okxProvider
```

Comment:
We use the same `context` object in the rest of the functions which use channels, should we use the same here? Wdyt?
At least, since we error here we should : 
```suggestion
		case config.ProviderOkx:
			priceProvider, err := provider.NewOkxProvider(context.TODO(), o.providerPairs[config.ProviderOkx]...)
			if err != nil {
				return nil, err
			}
```

Reply:
since we are returning 2 vars in the NewOkxProvider, it creates new variables 
So, to set it on the priceProvider it needs to have a different name 
Otherwise:
![image](https://user-images.githubusercontent.com/17556614/153631445-9ed43457-8c94-4372-8280-fb0635a810ca.png)

Reply:
👍 👍 any opinion on the `context` usage?

Reply:
In my idea, I was using the same context where we have channels to be able to stop it from grabbing new messages from outside of the `OkxProvider`

Reply:
Actually right now I can't think of a case where we would stop it, do you have any thoughts on this?

Reply:
like this? https://github.com/umee-network/umee/pull/522/commits/268ebaec4553ecfcfe8620ed852c43255a39c52a

Reply:
Yeah, although it might be good to init it separately from the tick to make it a bit cleaner. Wdyt about this @alexanderbez ?